orchestration/src/test/java/com/sap/ai/sdk/orchestration/OrchestrationUnitTest.java

```diff
@@ -129,34 +133,100 @@ void testCompletion() {
-    assertThat(groundingData.get("grounding_query")).isEqualTo("grounding call");
-    assertThat(groundingData.get("grounding_result").toString())
-        .startsWith("Joule is the AI copilot that truly understands your business.");
-    assertThat(result.getModuleResults().getGrounding().getMessage()).isEqualTo("grounding result");
-    assertThat(((ChatMessage) result.getModuleResults().getTemplating().get(0)).getContent())
-        .startsWith(
-            "What does Joule do? Use the following information as additional context: Joule is the AI copilot that truly understands your business.");
-    assertThat(llmChoice.getMessage().getContent())
-        .startsWith(
-            "Joule is an AI copilot that revolutionizes how users interact with their SAP business systems.");
-    assertThat(llmChoice.getFinishReason()).isEqualTo("stop");
-    assertThat(llmChoice.getMessage().getContent())
-        .startsWith(
-            "Joule is an AI copilot that revolutionizes how users interact with their SAP business systems.");
+
+    final var databaseFilter =
+        DocumentGroundingFilter.create()
+            .id("arbitrary-user-defined-id")
+            .dataRepositoryType(DataRepositoryType.VECTOR);
+    final var groundingConfigConfig =
+        GroundingModuleConfigConfig.create()
+            .inputParams(List.of("query"))
+            .outputParam("results")
+            .addFiltersItem(databaseFilter);
+    final var groundingConfig =
+        GroundingModuleConfig.create()
+            .type(GroundingModuleConfig.TypeEnum.DOCUMENT_GROUNDING_SERVICE)
+            .config(groundingConfigConfig);
+    final var configWithGrounding = config.withGroundingConfig(groundingConfig);
+
+    final Map<String, String> inputParams =
+        Map.of("query", "String used for similarity search in database");
+    final var prompt =
+        new OrchestrationPrompt(
+            inputParams,
+            Message.system("Context message with embedded grounding results. {{?results}}"));
+
+    final var response = client.chatCompletion(prompt, configWithGrounding);
+
+    assertThat(response).isNotNull();
+    assertThat(response.getOriginalResponse().getRequestId())
+        .isEqualTo("e5d2add4-408c-4da5-84ca-1d8b0fe350c8");
+
+    var moduleResults = response.getOriginalResponse().getModuleResults();
+    assertThat(moduleResults).isNotNull();
+
+    var groundingModule = moduleResults.getGrounding();
+    assertThat(groundingModule).isNotNull();
+    assertThat(groundingModule.getMessage()).isEqualTo("grounding result");
+    assertThat(groundingModule.getData()).isNotNull();
+    var groundingData =
+        Map.of(
+            "grounding_query",
+            "grounding call",
+            "grounding_result",
+            "First chunk```Second chunk```Last found chunk");
+    assertThat(groundingModule.getData()).isEqualTo(groundingData);
+
+    var systemMessage = (ChatMessage) moduleResults.getTemplating().get(0);
```

Comment:
These and the assertions below (except for the verify at the end) don't really add any coverage, as the same is already asserted in other test cases within this file. So I'd just remove it here